core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1014,16 +1014,19 @@ class SnappySession(@transient private val sc: SparkContext,
               }
             }
           case None =>
-            val r = DataSource(self,
+            val dataSource = DataSource(self,
               className = source,
               userSpecifiedSchema = userSpecifiedSchema,
               partitionColumns = partitionColumns,
-              options = params).write(mode, data)
-            if (None != userSpecifiedSchema) {
-              (r, Some(userSpecifiedSchema.get))
-            } else {
-              (r, Some(r.schema))
+              options = params)
+            val clazz: Class[_] = dataSource.providingClass
+            val r = clazz.newInstance() match {
+              case schemaRelationProvider: SchemaRelationProvider =>
+                schemaRelationProvider.createRelation(sqlContext, new CaseInsensitiveMap(params)
+                  , userSpecifiedSchema.getOrElse(data.schema))
```

Comment:
Thanks sumedh .I think for Column tables the ColumnarRelationProvider is used which implement SchemaRelationProvider so that will match here but will throw MatchError in other cases. I think option 'a' looks more safe.Will discuss once with Hemant about the ordering